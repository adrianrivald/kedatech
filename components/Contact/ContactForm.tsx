import React from "react";
import { SubmitHandler, useForm } from "react-hook-form";
import { Button } from "../Ui";
import { ToastContainer, toast } from "react-toastify";

export function ContactForm() {
  const form = React.useRef(null) as any;
  const { register, handleSubmit, watch, reset } = useForm<any>();
  const [isSubmitted, setIsSubmitted] = React.useState(false);

  const formValues = {
    name: watch("name"),
    email: watch("email"),
    message: watch("message"),
  };

  const onSubmit: SubmitHandler<any> = async () => {
    toast.success("Berhasil mengirim kontak, tunggu kabar selanjutnya", {
      position: "top-center",
      autoClose: 3000,
      hideProgressBar: true,
      closeOnClick: true,
      pauseOnHover: true,
      draggable: false,
      progress: undefined,
      theme: "light",
    });
    reset();
  };
  return (
    <form
      ref={form}
      onSubmit={handleSubmit(onSubmit)}
      className="flex flex-col gap-8"
    >
      <div className="flex flex-col gap-2">
        <label className="text-[14px]" htmlFor="name">
          Nama
        </label>
        <input
          role="inputNameField"
          id="name"
          {...register("name", { required: true })}
          type="text"
          className="rounded-md p-4 border border-neutral-100 focus:outline-none"
          placeholder="Masukkan nama Anda"
        />
      </div>
      <div className="flex flex-col gap-2">
        <label className="text-[14px]" htmlFor="email">
          Alamat Email
        </label>
        <input
          id="email"
          {...register("email", { required: true })}
          type="email"
          className="rounded-md p-4 border border-neutral-100 focus:outline-none"
          placeholder="Masukkan alamat email Anda"
        />
      </div>
      <div className="flex flex-col gap-2">
        <label className="text-[14px]" htmlFor="message">
          Pesan
        </label>
        <textarea
          id="message"
          {...register("message", { required: true })}
          className="rounded-md p-4 border border-neutral-100 focus:outline-none"
          placeholder="Masukkan pesan Anda"
          rows={4}
        />
      </div>
      <Button
        role="submitButton"
        isPrimary
        title="Kirim Pesan"
        className="lg:w-[250px]"
      />
    </form>
  );
}
