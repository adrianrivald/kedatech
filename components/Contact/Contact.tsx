import Image from "next/image";
import React from "react";
import { Fade } from "react-awesome-reveal";
import { twMerge } from "tailwind-merge";
import { maxWidthContainer } from "../../constants/class";
import { ContactForm } from "./ContactForm";

export function Contact() {
  return (
    <div
      id="contact"
      className={twMerge(
        "bg-[url('/assets/images/contact-bg.png')] p-4 lg:p-16 mt-16 lg:mt-24 bg-no-repeat bg-left bg-auto",
        maxWidthContainer
      )}
    >
      <div className="relative flex flex-col-reverse lg:flex-row flex  gap-16 backdrop-blur-lg border border-2 rounded-xl shadow-xl p-4 lg:p-16 relative min-h-[400px] w-[100%] mx-auto">
        <Fade direction="left" triggerOnce className="w-full lg:w-[50%]">
          <div className="">
            <ContactForm />
          </div>
        </Fade>
        <Fade direction="right" triggerOnce className="lg:w-[50%]">
          <div className="mt-12">
            <h2
              role="contactHeading"
              className="font-bold text-[24px] text-neutral-300"
            >
              Yuk tanya kami!
            </h2>
            <h3
              role="contactSubHeading"
              className="text-neutral-500 text-[32px] mt-4 font-bold"
            >
              Kirimkan Pertanyaan Anda pada Kami!
            </h3>
            <Image
              role="contactBg"
              src="/assets/images/contact-us.svg"
              width={612}
              height={408}
              alt="contact"
              className="hidden lg:block absolute right-2 bottom-2 w-1/2"
            />
          </div>
        </Fade>
      </div>
    </div>
  );
}
