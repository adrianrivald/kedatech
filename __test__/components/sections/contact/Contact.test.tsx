import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import renderer from "react-test-renderer";
import { Contact } from "../../../../components/Contact/Contact";

it("Benefit snapshot test", () => {
  const tree = renderer.create(<Contact />).toJSON();
  expect(tree).toMatchSnapshot();
});

beforeAll(() => {
  global.IntersectionObserver = class {
    constructor() {}
    observe() {}
    unobserve() {}
    disconnect() {}
  };
});

test("Contact element test", () => {
  render(<Contact />);

  const contactHeadingEl = screen.getByRole("contactHeading");
  expect(contactHeadingEl).toHaveTextContent("Yuk tanya kami!");

  const contactSubHeadingEl = screen.getByRole("contactSubHeading");
  expect(contactSubHeadingEl).toHaveTextContent(
    "Kirimkan Pertanyaan Anda pada Kami!"
  );

  const contactBgEl = screen.getByRole("contactBg");
  expect(contactBgEl).toHaveAttribute("src", "/assets/images/contact-us.svg");
});
