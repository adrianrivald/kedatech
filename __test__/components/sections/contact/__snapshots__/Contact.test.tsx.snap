// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Benefit snapshot test 1`] = `
<div
  className="bg-[url('/assets/images/contact-bg.png')] p-4 lg:p-16 mt-16 lg:mt-24 bg-no-repeat bg-left bg-auto max-w-[100rem] mx-auto"
  id="contact"
>
  <div
    className="relative flex flex-col-reverse lg:flex-row flex  gap-16 backdrop-blur-lg border border-2 rounded-xl shadow-xl p-4 lg:p-16 relative min-h-[400px] w-[100%] mx-auto"
  >
    <div
      className="w-full lg:w-[50%] css-0"
      style={
        {
          "animationDelay": "0ms",
          "opacity": 0,
        }
      }
    >
      <div
        className=""
        style={{}}
      >
        <form
          className="flex flex-col gap-8"
          onSubmit={[Function]}
        >
          <div
            className="flex flex-col gap-2"
          >
            <label
              className="text-[14px]"
              htmlFor="name"
            >
              Nama
            </label>
            <input
              className="rounded-md p-4 border border-neutral-100 focus:outline-none"
              id="name"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Masukkan nama Anda"
              role="inputNameField"
              type="text"
            />
          </div>
          <div
            className="flex flex-col gap-2"
          >
            <label
              className="text-[14px]"
              htmlFor="email"
            >
              Alamat Email
            </label>
            <input
              className="rounded-md p-4 border border-neutral-100 focus:outline-none"
              id="email"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Masukkan alamat email Anda"
              type="email"
            />
          </div>
          <div
            className="flex flex-col gap-2"
          >
            <label
              className="text-[14px]"
              htmlFor="message"
            >
              Pesan
            </label>
            <textarea
              className="rounded-md p-4 border border-neutral-100 focus:outline-none"
              id="message"
              name="message"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Masukkan pesan Anda"
              rows={4}
            />
          </div>
          <button
            className="rounded-lg px-4 py-2 border-2 min-h-12 bg-primary-500 border-primary-500 text-white lg:w-[250px]"
            role="submitButton"
          >
            Kirim Pesan
          </button>
        </form>
      </div>
    </div>
    <div
      className="lg:w-[50%] css-0"
      style={
        {
          "animationDelay": "0ms",
          "opacity": 0,
        }
      }
    >
      <div
        className="mt-12"
        style={{}}
      >
        <h2
          className="font-bold text-[24px] text-neutral-300"
          role="contactHeading"
        >
          Yuk tanya kami!
        </h2>
        <h3
          className="text-neutral-500 text-[32px] mt-4 font-bold"
          role="contactSubHeading"
        >
          Kirimkan Pertanyaan Anda pada Kami!
        </h3>
        <img
          alt="contact"
          className="hidden lg:block absolute right-2 bottom-2 w-1/2"
          data-nimg="1"
          decoding="async"
          height={408}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          role="contactBg"
          src="/assets/images/contact-us.svg"
          srcSet="/assets/images/contact-us.svg 1x, /assets/images/contact-us.svg 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={612}
        />
      </div>
    </div>
  </div>
</div>
`;
